File: src/day4/Main.java
package day4;

import java.io.File;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

class Main {
    public static void main(String[] args) throws Exception{
        File file = new File("/Users/williamfrisk/Documents/GitHub/AoC-2022/src/day4/puzzle_input.txt");
        Scanner sc = new Scanner(file);

        List<String> inputList = new ArrayList<>();

        while (sc.hasNextLine()) {
            inputList.add(sc.nextLine());
        }
        sc.close();

        System.out.println(task1(inputList));
    }

    private static int task1(List<String> arr) {
        return -1;
    }
}

File: src/day4/main.py

def task_1(arr: list) -> int:
    result = 0

    for pairs in arr:
        if (pairs[0] <= pairs[2]) and (pairs[1] >= pairs[3]):
            result += 1
        elif (pairs[0] >= pairs[2]) and (pairs[1] <= pairs[3]):
            result += 1

    return result


def task_2(arr: list) -> int:
    result = 0

    for pair in arr:
        if pair[2] <= pair[1] and pair[0] <= pair[3]:
            result += 1

    return result





if __name__ == "__main__":
    input_array = []
    with open('./src/day4/puzzle_input.txt') as f:
        for line in f:
            line = line[:-1]
            line = line.replace("-", ",")
            line = line.split(",")
            input_array.append([int(x) for x in line])

    print(task_2(input_array))
